CMakeLists.txt

```diff
@@ -153,23 +140,20 @@ endif()
 if(MSVC)
   list(APPEND MANIFOLD_FLAGS /DNOMINMAX /bigobj)
 else()
-  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    list(APPEND WARNING_FLAGS -Wall -Wno-unknown-warning-option -Wno-unused)
-  else()
-    list(
-      APPEND
-      WARNING_FLAGS
-      -Wall
-      -Wno-unknown-warning-option
-      -Wno-unused
-      -Wno-shorten-64-to-32
-    )
-  endif()
+  list(
+    APPEND
+    WARNING_FLAGS
+    -Wall
+    -Wno-unknown-warning-option
+    -Wno-unused
+    -Wno-shorten-64-to-32
+  )
```

Comment:
Are there compiler flags now supported across all supported versions of (gcc and clang) compilers?

Reply:
I think the most important flag is `-Wno-unknown-warning-option`, it allows us to use some other warning options not supported by the compiler. We should check when gcc and clang supports this...

Reply:
This seems to be a clang only flag, which is supported at least 13 years ago (https://bugzilla.mozilla.org/show_bug.cgi?id=731316).

GCC will not complain about unknown warning options anyway:

> When an unrecognized warning option is requested (e.g., -Wunknown-warning), GCC emits a diagnostic stating that the option is not recognized. However, if the -Wno- form is used, the behavior is slightly different: no diagnostic is produced for -Wno-unknown-warning unless other diagnostics are being produced. This allows the use of new -Wno- options with old compilers, but if something goes wrong, the compiler warns that an unrecognized option is present. 

These flags should be fine.